### ISSUE TITLE
[Windows 10] window size and position bug

### ISSUE THREAD

**Issue by MRN314** [telegramdesktop/tdesktop#1189]
After each maximize and restore telegram window size is reduced.
App randomly starts at desktop center (first monitor) instead of last saved position.

**Comment by DaFri-Nochiterov** [comment]
I can confirm first, everything works fine.
I have only one monitor, but after closing, TDesktop starts with the same position and size.

**Comment by MRN314** [comment]
Just tested again. Rebooted system 10 times and got 1 problem with window position restoring. Maybe it’s only windows 10 related problem, never happened on 8.1

**Comment by Brawl345** [comment]
I have the same problem with the current release, but I'm on Windows 7. I noticed the same bug on Windows 10 and it's really annoying.

**Comment by Ladislav92** [comment]
Go to folder option and check "Launch folder in separate process" 
![image](https://cloud.githubusercontent.com/assets/22495351/18932706/4f638958-85da-11e6-8c3c-20a1979ea1da.png)
/cloud.githubusercontent.com/assets/22495351/18932706/4f638958-85da-11e6-8c3c-20a1979ea1da.png)

**Comment by N0x1k** [comment]
This might be the same thing as https://github.com/telegramdesktop/tdesktop/issues/1693 Basically the issue is that Telegram doesn't remember the position when you use Aero snap (snap to side of screen) and then minimize/close to systray. If you minimize to task bar, it works, but once you turn off the task bar icon, it uses the last size and position before using the snap feature.

**Comment by Aokromes** [comment]
I confirm this, it needs few maximize+restore to notice.

**Comment by JohnKiller** [comment]
After build 1803 (April update) this bug is more frequend and also when window is maximized on the second monitor, the topbar is cut in half

**Comment by DaFri-Nochiterov** [comment]
As far as I understand, this issue was about resizing a window and that it changes its position after being restored. What you talking is a common bug with applications on Windows 10 (with April update and for a long time on Insider) that have custom title bars: when maximized, sometimes such windows can be wrongly maximized (means, the title bar is slightly cropped; also when you have a taskbar on left/right side, bottom also can be cropped). I'm not sure there's something to do with Telegram Desktop. I can reproduce the same thing with Discord, VK Messenger. Your commentary is probably irrelevant to this issue and should be reported as separate issue (needs confirmation by project maintainers).

The bug about Telegram window losing its position or size after restore isn't reproducible for me.

**Comment by lightlessfilms** [comment]
its a bit better if you adjust the settings under Compatibility tab in properties.
![2018-05-03 18_11_03-window](https://user-images.githubusercontent.com/7137900/39589035-5efdda7e-4efd-11e8-9be9-d516d54a6e1e.png)

**Comment by bendbot** [comment]
Скорей всего проблема появляется у тех, кто использует больше одного монитора.
Если второй монитор находится слева от основного, то расположение окна в левой части основного монитора рассчитывается неправильно и приводит к сбросу при следующем открытии программы.
Если расположить второй монитор справа от основного, то все работает нормально.

Изменение [параметров совместимости](https://github.com/telegramdesktop/tdesktop/issues/1189#issuecomment-386349795) ни к чему не приводит.
Изменение [Параметра папки "Запускать окна с папками в отдельном процессе"](https://github.com/telegramdesktop/tdesktop/issues/1189#issuecomment-250304006) тоже ничего не дает.

Записал видео как ведет окно Telegram https://youtu.be/4cCg0uQQecQ

**Comment by Vilmodao** [comment]
I must have something like that.
Every time I restart Telegram its window comes smaller. Looks like it loose exactly 16 pix. by width in one go. Maximize and restore goes well.
Took screenshots after every restart.
![Untitled-1](https://user-images.githubusercontent.com/11928504/75109637-2bb92c80-5636-11ea-958f-3795376faacc.gif)
Windows 8.1 (not highly updated)
Telegram Desktop Portable v 1.9.12 (it seems the problem was on the previous version as well)

**Comment by lightlessfilms** [comment]
> I must have something like that.
> Every time I restart Telegram its window comes smaller. Looks like it loose exactly 16 pix. by width in one go. Maximize and restore goes well.
> Took screenshots after every restart.
> https://user-images.githubusercontent.com/11928504/75109637-2bb92c80-5636-11ea-958f-3795376faacc.gif
> Windows 8.1 (not highly updated)
> Telegram Desktop Portable v 1.9.12 (it seems the problem was on the previous version as well)
> 
> Oh, I didn't look at the date. Maybe I should create a new ticket?

Please do :) Also, what version of Windows 8.1 do you use?

**Comment by Aokromes** [comment]
> I must have something like that.
> Every time I restart Telegram its window comes smaller. Looks like it loose exactly 16 pix. by width in one go. Maximize and restore goes well.
> Took screenshots after every restart.
> [user-images.githubusercontent.com/11928504/75109637-2bb92c80-5636-11ea-958f-3795376faacc.gif](https://user-images.githubusercontent.com/11928504/75109637-2bb92c80-5636-11ea-958f-3795376faacc.gif)
> Windows 8.1 (not highly updated)
> Telegram Desktop Portable v 1.9.12 (it seems the problem was on the previous version as well)
> 
> Oh, I didn't look at the date. Maybe I should create a new ticket?

no need of create new tickets for already opened issues even they are ancient.

**Comment by Vilmodao** [comment]
> Please do :) Also, what version of Windows 8.1 do you use?

6.3.9600 Build 9600 


> no need of create new tickets for already opened issues even they are ancient.

Sure. But it seems to me, the closed ticket #7222 is more relevant to my case.

**Comment by lightlessfilms** [comment]
@Vilmodao Thank you for the quick reply. Can you install the latest version of Telegram Desktop portable version? 1.9.14 is out

**Comment by Vilmodao** [comment]
>  Thank you for the quick reply. Can you install the latest version of Telegram Desktop portable version? 1.9.14 is out

Oops... I already use 1.9.14. I wrote 1.9.12 by mistake. Sorry. I just fixed it.

**Comment by nasaboy** [comment]
The size of the window will shrink automatically, each time it is closed and then opened again, until it shrinks to a certain extent.
1.9.19 also the same.

**Comment by alexthrasher** [comment]
> Sure. But it seems to me, the closed ticket #7222 is more relevant to my case.

Same here. I was just about to create a post about this issue when I realized it has tons of duplicates, but the #7222 is more relevant to me as well

> The size of the window will shrink automatically, each time it is closed and then opened again, until it shrinks to a certain extent.
1.9.19 also the same.

I'm also running 1.9.19 beta, and it has been happening since a few updates ago on Windows 10. And it's quite annoying. It happens specially when you drag to resize until it snaps to bottom or top of the screen, otherwise it still remembers window size but only if you don't completely close the application

**Comment by nasaboy** [comment]
> > Sure. But it seems to me, the closed ticket #7222 is more relevant to my case.
> 
> Same here. I was just about to create a post about this issue when I realized it has tons of duplicates, but the #7222 is more relevant to me as well
> 
> > The size of the window will shrink automatically, each time it is closed and then opened again, until it shrinks to a certain extent.
> > 1.9.19 also the same.
> 
> I'm also running 1.9.19 beta, and it has been happening since a few updates ago on Windows 10. And it's quite annoying. It happens specially when you drag to resize until it snaps to bottom or top of the screen, otherwise it still remembers window size but only if you don't completely close the application

the 1.9.21 also have the problem,it's quite annoying.so sad.

**Comment by Vilmodao** [comment]
The window shrinking bug is still here on version 2.1.

**Comment by nasaboy** [comment]
2.1.6 not fixed

**Comment by mmortal03** [comment]
Yep. My Telegram window height gets smaller after each Telegram restart. Telegram version 2.1.13. Windows 10 version 1909. Laptop screen resolution 1600x900.

**Comment by ezequielfalcon** [comment]
Same here.
2.1.14 and Windows 10 2004

**Comment by nasaboy** [comment]
2.2.0 not fixed hhh

**Comment by Vilmodao** [comment]
2.3.1 not fixed

**Comment by nasaboy** [comment]
windows update to 2004,it fixed.

**Comment by alexbraga** [comment]
> windows update to 2004,it fixed.

I'm on a fresh Windows install, recently updated to 2004 and the bug persists.

**Comment by last-things** [comment]
Apparently the problem can occur when you run the program minimized by default - setting it to launch normally fixed it for me. Still, would be nice to have that option without the window size bugging out. For a long time, I used to think the resize was an intentional change on Telegram's part.

**Comment by last-things** [comment]
It broke again, either after Tg or Windows update. I don't use multiple monitors or virtual desktops.

**Comment by mmortal03** [comment]
@last-things , I noticed that the Telegram window currently can't physically be shrunk vertically past a certain point. So, for me, it gets smaller and smaller vertically until it hits that point, and then obviously no longer happens because it's reached that point. It's still not fixed for people who want the window to be larger than that, though.

**Comment by shemanaev** [comment]
"Use system window frame" seems to fixed this. 
The right sizes written and read according to debug log, so bug is somewhere in applying size to custom title bar/shadow.

**Comment by mmortal03** [comment]
> "Use system window frame" seems to fixed this.

Thanks, that does seem to be a workaround.

**Comment by Younes-L** [comment]
the "use system window frame" workaround doesn't totally fix this for me only when using vertical right panel (aka dock) as is default on ubuntu.
It's as if tdesktop tries to avoid a non-existent right panel. Only happens in this position and not the others, in other panel positions, tdesktop launches with correctly saved position from previous launch (no space between tdesktop's window and system panel)
I'm using Mint 20.1 with Cinnamon 4.8.5 , should i open a new issue ?

**Comment by Aokromes** [comment]
https://betanews.com/2021/04/29/microsoft-is-fixing-multiple-monitor-support-in-windows-10/

**Comment by mmortal03** [comment]
> https://betanews.com/2021/04/29/microsoft-is-fixing-multiple-monitor-support-in-windows-10/

That's great, but it makes no sense to close this issue based on that. The people experiencing this bug aren't inherently on multiple monitors, and the bug isn't some general issue with Windows programs -- it only affects Telegram. I can still demonstrate the bug for you right now.

**Comment by Aokromes** [comment]
> > [betanews.com/2021/04/29/microsoft-is-fixing-multiple-monitor-support-in-windows-10](https://betanews.com/2021/04/29/microsoft-is-fixing-multiple-monitor-support-in-windows-10/)
> 
> That's great, but it makes no sense to close this issue based on that. The people experiencing this bug aren't inherently on multiple monitors, and the bug isn't some general issue with Windows programs -- this issue only affects Telegram. I can still demonstrate the bug for you right now.

i am unable to reproduce on latest windows insider with a single monitor.

**Comment by Aokromes** [comment]
> > i am unable to reproduce on latest windows insider with a single monitor.
> 
> I'm working on reproducing it on another machine for you to help determine the specific conditions. It still does it on my laptop, but not on a desktop. Both are using the 32-bit version of Telegram Desktop, version 2.7.4. I'm not on a Windows Insider branch, so I can't speak to that.
> 
> My laptop screen is set at 1600x900 resolution, with Windows scale set to 100%, and I've tried Telegram scale set to 100% or 125%, but it doesn't matter, it still does it.
> 
> The desktop monitor is set at 1920x1080 resolution, but I've also tried 1600x900, I've tried the Windows scale at both 100% and 125% at both 1920 and 1600 resolutions, and I've tried Telegram scale set to 100% or 125%, but it doesn't matter, I can't reproduce it there.
> 
> I had a hunch that it had something to do with differences in system menu font size or scrollbar size, but I compared various settings for those across both machines, controlled for it, and still couldn't reproduce on the desktop, only on the laptop.

i tried on the system where i reproduced it on 2016, i cannot now, so, microsoft fixed it.

**Comment by Aokromes** [comment]
10.0.21376.1

**Comment by mmortal03** [comment]
Before taking that step, I tested it on a clean install of 20H2 on the same laptop and couldn't reproduce. This got me thinking what it could be on my particular Windows install that was causing it. I decided to clean house, uninstalling an assortment of old packages and libraries that were still present but that I hadn't used in a while, including Java, MATLAB, Qt, gstreamer-sdk, etc., then rebooted the laptop, and the problem disappeared! 
I could possibly revert back to a previous backup image to try to determine which item was specifically the cause, but I don't think it's worth the effort -- people still experiencing the bug should just try to uninstall various older libraries that they still have present on their machine and reboot and see if it fixes it. Maybe one of the devs here could point to applicable libraries or packages that would be able to interfere with Telegram in this way? My first thought is the very old copy of Qt that I had installed.
